### ISSUE TITLE
bug: false positive for proper_super_calls

### ISSUE THREAD

**Issue by yurii-prykhodko-solid** [solid-software/solid_lints#235]
<img width="537" height="227" alt="Image" src="https://github.com/user-attachments/assets/bde98fd6-584b-4749-8645-4d2b0ab6c750" />

```
<filename>.dart:137:3 • super.dispose() should be
      last • proper_super_calls • INFO
```

**Comment by yurii-prykhodko-solid** [comment]
We probably don't handle async methods properly in the rule.

**Comment by solid-illiaaihistov** [comment]
Fixed
